### PR TITLE
test: add pod e2e testing

### PR DESCRIPTION
**Pull request by kuangcheng66**
#### What type of PR is this?
test: add test codes

#### What does this PR do?
add pod e2e testing code

#### Screenshots (if applicable):
```

```

#### Additional Comments:
```

```

### CODE REVIEW

**Comment by kuangcheng66** [comment]
@zhangyongxi
@zhouhaoA1    please help review.

File: test/e2e/framework/pod.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	"github.com/kosmos.io/kosmos/pkg/generated/clientset/versioned"
+)
```
Comment (duanmengkk) [inline test/e2e/framework/pod.go]: Is the code haved been formated?

File: test/e2e/framework/pod.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// CreatePod create Pod.
+func CreatePod(client kubernetes.Interface, pod *corev1.Pod) {
+	ginkgo.By(fmt.Sprintf("Creating Pod(%s/%s)", pod.Namespace, pod.Name), func() {
+		_, err := client.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// RemovePod delete Pod.
```
Comment (duanmengkk) [inline test/e2e/framework/pod.go]: We can delete the  unuseless comment

File: test/e2e/framework/pod.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	"github.com/kosmos.io/kosmos/pkg/generated/clientset/versioned"
+)
+
+// CreatePod create Pod.
```
Comment (duanmengkk) [inline test/e2e/framework/pod.go]: We can delete the unuseless comment or make the comment longer

File: test/e2e/framework/pod.go (hunk 2 above)
Comment (kuangcheng66) [reply test/e2e/framework/pod.go]: ok，I will fix it.

File: test/e2e/framework/pod.go (hunk 3 above)
Comment (kuangcheng66) [reply test/e2e/framework/pod.go]: ok

File: test/e2e/suit_test.go [hunk 4]
```diff
@@ -42,9 +42,9 @@ func init() {
 	// eg. ginkgo -v --race --trace --fail-fast -p --randomize-all ./test/e2e/ -- --poll-interval=5s --pollTimeout=5m
 	flag.DurationVar(&pollInterval, "poll-interval", 5*time.Second, "poll-interval defines the interval time for a poll operation")
 	flag.DurationVar(&pollTimeout, "poll-timeout", 300*time.Second, "poll-timeout defines the time which the poll operation times out")
-	flag.StringVar(&hostContext, "host-context", "kind-cluster-host", "name of the host cluster context in kubeconfig file.")
```
Comment (zhouhaoA1) [inline test/e2e/suit_test.go]: I think we should keep it, just rewrite default value from `kind-cluster-host` to `current-context`.

File: test/e2e/suit_test.go [hunk 5]
```diff
@@ -55,13 +55,18 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
...
 	gomega.Expect(kubeconfig).ShouldNot(gomega.BeEmpty())
+
 	config, err := framework.LoadRESTClientConfig(kubeconfig, hostContext)
 	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
 	restConfig = config
 	kubeClient, err = kubernetes.NewForConfig(restConfig)
 	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-	clusterLinkClient, err = versioned.NewForConfig(restConfig)
-	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
```
Comment (zhouhaoA1) [inline test/e2e/suit_test.go]: suggest keep it

File: test/e2e/suit_test.go (hunk 4 above)
Comment (kuangcheng66) [reply test/e2e/suit_test.go]: ok. what you said is right.

File: test/e2e/suit_test.go (hunk 5 above)
Comment (kuangcheng66) [reply test/e2e/suit_test.go]: kosmosclient and clusterlinkclient are the same, it is better to use the former uniformly.

File: test/e2e/framework/pod.go (hunk 1 above)
Comment (kuangcheng66) [reply test/e2e/framework/pod.go]: ok,no problem

**Comment by kuangcheng66** [comment]
@xyz2277 @duanmengkk @zhouhaoA1  Please help merge.

**Pull request closed without merging**